specs/Deployment/MSIXPackageVersioning.md

```diff
@@ -0,0 +1,511 @@
+
+|Option|Syntax|Example|Information|Comments|
+| --- | :--- | :---: | :---: | :---: |
+|A|X.N.P.0|0.8.1.0<br>1.2.3.0<br>17.14.3.0|Major Minor Patch||
+|B|X.NYYMM.DDPPP.0|0.82106.04001.0<br>1.22112.31003<br>--error--|Major Minor Patch Date|Minor max is 6<br>Patch max is 999<br>WinUI 2.x used this encoding<br> Windows App SDK 0.5 used this encoding|
+|C|XNN.PYYMM.DDBBB.0|8.12106.04123.0<br>102.32112.31123.0<br>1714.33110.20123.0|Major Minor Patch Date Build|Minor max is 99<br>Patch max is 6<br>Build max is 999|
+|D|NPPP.E.B.0|8001.154.123<br>2003.364.123<br>17014.3944.123.0|Minor Patch Elapsed Build|Minor max is 64<br>Patch max is 999<br>Build max is 65535|
+|E|NPP.E.B.0|801.154.123<br>203.364.123<br>1714.3944.123.0|Minor Patch Elapsed Build|Minor max is 99<br>Patch max is 99<br>Build max is 65535|
+
+***Recommendation:*** Option D with Epoch=January 1, 2021.
+
+Option A works for a simple release strategy but doesn't work when there's regular
+(e.g. daily) builds.
+
+Option B worked for WinUI 2.x and Windows App SDK 0.5 but is too limiting given it can't support
+Minor versions beyond 6. It's also a complicated encoding scheme with date injected between Minor
+and Patch (not readily legible).
+
+Option C is too limiting given it can't support Patch versions beyond 6.
+
+Option D relies on the package Name including Major version. This leaves sufficient space in the
+version to encode up to 99 Minor and 99 Patch versions. Date encoding as an Epoch reduces the size
+while supporting 20+ years with only 4 digits.
+
+Option E is a variant of Option D except Patch version is restricted to 2 digits, but it's an
+unnecessary simplification.
+
+### 2.4.1. Windows App SDK 0.5 Notation
+
+Windows App SDK 0.5 encodes MSIX versions as `M.NYYMM.DDPPP.0`
+
+* M = Major version
+* N = Minor version
+* YY = Year
+* MM = Month
+* DD = Day
+* PPP = Patch
+
+All fields are fixed-size with leading zeroes as needed. For example, Windows App SDK 0.5.0's MSIX package has a version of `0.52103.25000.0`
+
+* Major = 0
+* Minor = 5
+* Year = 21 (aka 2021)
+* Month = 03
+* Day = 25
+* Patch = 000 (if we have a 0.5.1 release this would be `001`)
+
+This doesn't scale going forward e.g. MSIX version is a DotQuadNumber aka `uint16[4]`. If 0.8 was released on May'31, 2021 MSIX version would be `0.82105.31000.0`
+
+### 2.4.2. WinUI 2.x Notation
+
+WinUI 2.x encodes MSIX versions as `M.NYYMM.DDPPP.0` (same as Windows App SDK 0.5). This runs into issues starting with WinUI 2.7, as identified by [WinUI version schema sometime can't applied to MinVersion #4008](https://github.com/microsoft/microsoft-ui-xaml/issues/4008#issuecomment-765633788).
+
+## 2.5. Decision 5: Package Names
+
+Windows App SDK 0.8+ has 3 MSIX packages. We have choices for package Name patterns for 0.8 and 1.x.
+
+Assuming Breaking Change Boundary for v0.x and v1+'s decision are `Major.Minor` version we have the
+following naming patterns...
+
+* WASfwk: `Microsoft.WindowsAppRuntime.<rmajor>.<rminor>[-tag]`
```

Comment:
How do <rmajor> and <rminor> differ from <major> and <minor>?

Reply:
rmajor+rminor are the Release version, e.g. 0 nd 8 for what was ProjectReunion 0.8
major+minor are the 1st 2 parts of the MSIX package version

I see rmajor+rminor are missing from the bulleted list @ line 406. Thought I'd added that. Fixing